src/macros/macrorecorder.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <gtest/gtest_prod.h>
+
+#include <QDebug>
+#include <QObject>
+#include <QtCore>
+
+#include "control/controlpushbutton.h"
+#include "engine/channelhandle.h"
+#include "macros/macro.h"
+#include "recording/recordingmanagerbase.h"
+
+// TODO(xerus) add enum for recording_status
+
+enum MacroRecordingState : uint8_t {
+    /// Nothing is going on
+    Disabled,
+    /// Recording is active, but nothing currently going on
+    Armed,
+    /// An Action is currently being recorded
+    Recording,
+};
+
+/// The MacroRecorder handles the recording of Macros and the [MacroRecording] controls.
+class MacroRecorder : public RecordingManagerBase {
+    Q_OBJECT
+  public:
+    static const QString kControlsGroup;
+
+    MacroRecorder();
+
+    void startRecording() override;
+    void stopRecording() override;
+    bool isRecordingActive() const override;
+
+    /// Tries to append this cue jump to the currently recorded Macro.
+    /// Returns true if the currently recorded Macro was changed - so only if
+    /// recording is active and the channel handle matches.
+    /// Only called from RT.
+    void notifyCueJump(ChannelHandle& channel, double origin, double target);
```

Comment:
Ah, forgot about the usage of const

Reply:
Damn, can't use a const reference because I need to assign it. So I am using a pointer now, even though it doesn't have to be mutable, is that still correct?

What's the practical difference between reference and pointer in this context?

Reply:
This is mostly convention. Non-const references should be avoided because it looks like a regular pass-by-value from the caller side.